code/visualization_app.py

```
@@ -7,118 +7,178 @@
 from skimage import measure
 import matplotlib.colors as mcolors
 from io import BytesIO
+from typing import Optional, List, Tuple, Union
```

Ruff

`typing.Union` imported but unused

Remove unused import: `typing.Union`

(F401)